PR builds fail when ci-build-image is being rebuild

```
14:49:11  Authentication to docker.elastic.co...
14:49:11  ../hack/docker-push.sh: line 37: DOCKER_LOGIN: unbound variable
14:49:11  Makefile:56: recipe for target 'ci-build-image' failed
14:49:11  make[1]: *** [ci-build-image] Error 1
```
it looks like the `.registry.env` file is not set up properly (?) 

https://devops-ci.elastic.co/job/cloud-on-k8s-pr/3641/console
https://devops-ci.elastic.co/job/cloud-on-k8s-pr/3621/console

--- a/hack/docker-push.sh
+++ b/hack/docker-push.sh
@@ -12,13 +12,13 @@
 set -euo pipefail
 
 SCRIPT_DIR="$(cd "$(dirname "${BASH_SOURCE[0]}")" && pwd)"
-
+REGISTRY_ENV="$SCRIPT_DIR/../.registry.env"
 retry() { "$SCRIPT_DIR/retry.sh" 5 "$@"; }
 
 # source variables if present
-if [[ -f .registry.env ]]; then
+if [[ -f ${REGISTRY_ENV} ]]; then
     # shellcheck disable=SC2046
-    export $(sed "s|[[:space:]]*=[[:space:]]*|=|g" .registry.env)
+    export $(sed "s|[[:space:]]*=[[:space:]]*|=|g" "${REGISTRY_ENV}")
 fi
 
 docker-login() {

--- a/hack/docker-push.sh
+++ b/hack/docker-push.sh
@@ -13,6 +13,7 @@ set -euo pipefail
 
 SCRIPT_DIR="$(cd "$(dirname "${BASH_SOURCE[0]}")" && pwd)"
 REGISTRY_ENV="$SCRIPT_DIR/../.registry.env"
+
 retry() { "$SCRIPT_DIR/retry.sh" 5 "$@"; }
 
 # source variables if present

